zebra-consensus/src/error.rs

```diff
@@ -53,7 +53,16 @@ pub enum BlockError {
     #[error("invalid difficulty threshold in block header {0:?} {1:?}")]
     InvalidDifficulty(zebra_chain::block::Height, zebra_chain::block::Hash),
 
-    #[error("block {0:?} failed the difficulty filter: hash {1:?} must be less than or equal to the difficulty threshold {2:?}")]
+    #[error("block {0:?} failed the difficulty limit: the difficulty threshold {2:?} must be at least as difficult as the {3:?} difficulty limit {4:?}, block hash {1:?}")]
```

Comment:
I want to properly document the style I've been using with error messages but I don't know how to describe the change in sentence structure here, it's been too long since I had an English class. Typically though I try to avoid things like "failed" in the error message, and instead focus on describing what condition prevented the function from completing. eg instead of "failed to find directory" I'd write "directory not found".

```suggestion
    #[error("block {0:?} has difficulty threshold {2:?} that is below the {3:?} difficulty limit {4:?}")]
```

Then I think we should change the way we identify blocks to be consistent across the codebase, I think I added a `Display` impl that includes the height and the hash, so you may be able to just use that here now.

Reply:
The `Display` change is intrusive, and I've already based another branch on this PR.

So I opened #1158 for that change.

Reply:
oh, that sounds perfect, I didn't mean for the display change to be done as part of this PR, other than perhaps in this one error message.